Public/Src/Sandbox/Windows/DetoursServices/SendReport.cpp

```diff
@@ -81,7 +81,11 @@ void ReportFileAccess(
     else {
         fileName = policyResult.GetCanonicalizedPath().GetPathString();
     }
-
+    
+    if (wcspbrk((wchar_t*)fileName, L"\u000A\u000B\u000C\u000D\u0085\u2028\u2029")) {
+        fileName = L"";
+    }

```

Comment:
For verification, please add a unit test. You can add a unit test in [PipExecutorDetoursTest.cs](https://github.com/microsoft/BuildXL/blob/master/Public/Src/Engine/UnitTests/Processes.Detours/PipExecutorDetoursTest.cs); see for example this [test](https://github.com/microsoft/BuildXL/blob/4663518d2d9afb8ae9884b768816b596054ce176/Public/Src/Engine/UnitTests/Processes.Detours/PipExecutorDetoursTest.cs#L6348)

The test harness itself is in this [DetoursTest folder](https://github.com/microsoft/BuildXL/tree/5248abbde6962043288ed353f4a9738825c0dc89/Public/Src/Sandbox/Windows/DetoursTests)

Reply:
I worked on it for some time but I am struggling with the unit test, as I don't have a thorough knowledge of the codebase. I was wondering if it would be possible for someone from the BuildXL team to look at this, now that the issue has been identified?
Thank you

Reply:
Thanks for reporting this to us and helping with the fix! I will pick up this PR for you to make escape character changes and add the unit test.

Reply:
Sorry about the wait @DamGouz! I have pushed a fix for this issue for you: https://github.com/microsoft/BuildXL/commit/e58b9c97323d7620096207cc501a9c689389e2ec

Thank you for reporting this bug to us!